src/interface/src/app/password-reset/confirmation-dialog/confirmation-dialog.component.spec.ts

```diff
@@ -0,0 +1,75 @@
+import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
+import { TestbedHarnessEnvironment } from '@angular/cdk/testing/testbed';
+import { HarnessLoader } from '@angular/cdk/testing';
+import { ComponentFixture, TestBed } from '@angular/core/testing';
+import { MatButtonHarness } from '@angular/material/button/testing';
+import { MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { Router } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+import { MaterialModule } from 'src/app/material/material.module';
+
+import { ConfirmationDialogComponent } from './confirmation-dialog.component';
+
+describe('ConfirmationDialogComponent', () => {
+  let component: ConfirmationDialogComponent;
+  let fixture: ComponentFixture<ConfirmationDialogComponent>;
+  let loader: HarnessLoader;
+  const routerStub = () => ({ navigate: (_: string[]) => ({}) });
+
+  beforeEach(async () => {
+    const fakeDialog = jasmine.createSpyObj(
+      'MatDialog',
+      {
+        open: undefined,
```

Comment:
these are methods, maybe its better to mock them like `open: () => {}` and `close: () => {}`

Reply:
Done. Thanks